server/cluster.go

```diff
@@ -523,6 +523,21 @@ func (c *Cluster) isRemoteTopic(topic string) bool {
 	return c.ring.Get(topic) != c.thisNodeName
 }
 
+// genLocalTopicName is just like genTopicName(), but the generated name belongs to the current cluster node.
+func (c *Cluster) genLocalTopicName() string {
+	topic := genTopicName()
+	if c == nil {
+		// Cluster not initialized, all topics are local
+		return topic
+	}
+
+	// FIXME: if cluster is large it may become too inefficient.
+	for c.ring.Get(topic) != c.thisNodeName {
```

Comment:
Humm... would it be possible to generate the name on the node where the original session sits? This way, we'll land on the correct node from the very start.

Reply:
This is exactly that. It generates topic name on the origin node making sure the new topic is hosted on the current node.